docs/alerts.md

```diff
@@ -262,3 +262,51 @@ handler2 = (tables=<-) => slack.notify(channel: "#flux")
 tables |> handler1() |> to(bucket: "notifications")
 tables |> handler2() |> to(bucket: "notifications")
 ```
+
+### Tasks
+
+Using the above, this can be integrated with tasks fairly easily and meet the above requirements.
+
+This might be the check script.
+
+```
+// Injected.
+option task = {
+	every: 1m,
+}
+
+// Possibly an option like now?
+notify = (tables=<-) => tables |> to(bucket: "system")
```

Comment:
Why is this function called `notify`? All it does is write to the system bucket.

Reply:
Because it abstracts away how the notifications are implemented. So you would put `notify()` in your script to say "I want to send a notification" and we would insert the notification behavior. Notify can also be implemented as an option too.

Reply:
The design in the SPEC is that checks and notifications are loosely coupled. This design pattern forces checks and notifications to be tightly coupled because a check's status is not persisted unless a `notify` function is called.

We need to allow for checks to be persisted to a bucket without any call to notify functions.

Reply:
I think you and I might just disagree on this. I do not see how they are tightly coupled here. The notify just signals that a notification should happen based on the check result and the implementation of notify is completely abstracted away. It could either be the actual implementation so the notification happens in the same task or it could be part of a notification task as shown below.

Is this a problem with the syntax that is used? The reason why I like this pattern is because it is easier to understand the intention of the user.

Reply:
What I mean by tightly coupled is that in this design in order for a check's status to be persisted it must be passed to an implementation of a notify function. We do not want users to think that every check must have a notification. Users should be able to create checks and not notify on them. An example workflow is  developing a new check. The users does the following: write the check, leave it running for a while, manually inspect the statuses of the check, repeat until satisfied by its behavior, and finally define a notification that matches the check. 

From the spec,

>Checks and notifications are separate and distinct. Checks can be delivered first and then notifications can be introduced next

> Notification Rules are loosely coupled to checks by the matching check tags. One notification rule may match several checks.

Reply:
I'll integrate this in. Thanks for the explanation. I think I understand it better now.

Reply:
This is now integrated. I think something in the notifications part needs to change, but I want you to take a look at the checks part first before I try to integrate a similar solution into notifications or come up with something different there.

Reply:
Check parts look good